framework/src/utils/MooseUtils.C

```diff
@@ -54,10 +54,13 @@ escape(std::string & str)
 }
 
 std::string
-trim(std::string str, const std::string & white_space)
+trim(const std::string & str, const std::string & white_space)
 {
-  std::string r = str.erase(str.find_last_not_of(white_space) + 1);
-  return r.erase(0, r.find_first_not_of(white_space));
+  const auto begin = str.find_first_not_of(white_space);
+  if (begin == std::string::npos)
+    return ""; // no content
+  const auto end = str.find_last_not_of(white_space);
```

Comment:
Wait, what if end is std::string::npos? I'm pretty sure doing arithmetic on that will fail.

Reply:
I played around with this and can't get it to fail, including adding a trim of space only and an empty string. I even put a mooseError on with end==npos and it never gets hit. I think that 62 will returns the beginning iterator in those cases, which makes 63 work like it should.